lib/chef/resource/rhsm_subscription.rb

```diff
@@ -32,11 +32,19 @@ class RhsmSubscription < Chef::Resource
         name_property: true
 
       action :attach, description: "Attach the node to a subscription pool." do
+        already_attached = subscription_attached?(new_resource.pool_id)
```

Comment:
This would thing could be even cleaner if we just killed off the execute
```
unless already_attached
  converge_by("attach subscription pool #{new_resource.pool_id}") do
    shell_out!("subscription-manager attach --pool=#{new_resource.pool_id}")
  end
end
```

That way the no-op doesn't show the execute in the logs. That's totally untested, but *should* work if the current stuff here works